### ISSUE TITLE
[BUG] Memory Leak on google-collab with nvstring column + string-groupby  

### ISSUE THREAD

**Issue by VibhuJawa** [rapidsai/cudf#2417]
**Describe the bug**
CPU Memory Leak on collab with  nvstrings col assignment + string-groupby  

The RAM used after  column assignment + groupby increases with each function call. 

This occurs only on  `Google-Collab` . On my native install, the memory used does not increase by that much.  

**Steps/Code to reproduce bug**
Collab Link: 

https://colab.research.google.com/gist/VibhuJawa/585d5981c5e50be2fa67a6df5451b00a/collab_rapids_mem_test.ipynb

```python
import cudf
import nvstrings
import numpy as np
import gc
import nvtext


def do_string_groupby(big_string):
    """
        returns None
    """ 
    df = cudf.DataFrame()
    # memory is not used if directly assigning to custring series
    # df['string'] = big_string
    df['string'] = big_string.data
    df['counts'] = np.dtype('int32').type(0)
    
    res = df.groupby('string').count()

    return None

big_string = cudf.Series(['one two three']*10_000_000)

!free --human

for _ in range(0,20):
    do_string_groupby(big_string)

gc.collect()
!free --human
```

Ouput:
Before running the function

```
             total        used        free      shared  buff/cache   available
Mem:            12G        1.3G          9G        6.9M        1.4G         11G
Swap:            0B          0B          0B
```
After running the function
```
              total        used        free      shared  buff/cache   available
Mem:            12G        2.5G        8.9G        6.9M        1.3G         10G
Swap:            0B          0B          0B
```

**Expected behavior**
Expected behavior would be not use this much memory. 

On a non-collab environment the memory used  only increases by 100 MB. 

```
              total        used        free      shared  buff/cache   available
Mem:           376G        7.3G         53G        157M        315G        367G
Swap:          6.0G        100M        5.9G

```

```
              total        used        free      shared  buff/cache   available
Mem:           376G        7.4G         53G        157M        315G        367G
Swap:          6.0G        100M        5.9G
```

**Environment overview (please complete the following information)**
 - Environment location: `google collab`

Edit: Edit to reflect issue not  with tokenize but with assigning a column with `nvstrings` .

**Comment by kkraus14** [comment]
@VibhuJawa can you reproduce without `tokenize`? Can you reproduce with just `tokenize` without the groupby?

**Comment by VibhuJawa** [comment]
@kkraus14 , editting this to confirm:


Can reproduce it without tokenize, if i assign an `nvstrings instance` instead of a `cudf string series`:

See below:
Ram is used if:
```python
def do_string_groupby(big_string):
    """
        returns None
    """ 
    df = cudf.DataFrame()
    # memory is not used if directly assigning to cudf string series
    # df['string'] = big_string
    df['string'] = big_string.data
    df['counts'] = np.dtype('int32').type(0)
    
    res = df.groupby('string').count()
    return None
```

**Comment by devavret** [comment]
Is this still an issue after the groupby refactor in #4346

**Comment by kkraus14** [comment]
This is fixed by #4346
